src/Blue/Tools/Api/Client.php

```diff
@@ -156,20 +156,29 @@ private function resolve(ResponseInterface $response)
 
             $attempts = $this->deferredResultMaxAttempts;
 
+            $locationHeader = $response->getHeader('location');
+
             while ($attempts > 0) {
+
+                if($locationHeader) {
```

Comment:
Would it be safer to make the `get_deferred_results` API responses return a location header as well and create a single codepath for deferrals so that we don't worry about false positives? @davvblack?